.NET rewrite (#30)
helldivers-2/api: CONTRIBUTING.md

that we are open to implementing the feature
(see [Why we might not accept all Pull Requests](#why-we-might-not-accept-all-pull-requests)).
Once a maintainer (most notably @dealloc and/or @lambstream) approves your change request, you can get started.

## Fork the API
See [Github's fork documentation](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo).

## Set up a local development environment
(TODO: expand on all required dependencies, setup etc)
Once you have cloned your fork or the API itself, run the following commands to initialize the local environment:
```shell
git submodule update .\src\Helldivers-2-Models\json
dotnet restore
dotnet build --no-restore --no-incremental
```
If you don't get any errors, that means your development environment is all set to get developing!
Open the `Helldivers-2.sln` file in the root of the repository and you're good to go!

### Formatting, line endings and syntax
There's three pilars when it comes to coding conventions, formatting etc:
- [.editorconfig](https://editorconfig.org/) is set up in the repository, if your IDE supports it you should install

>     the required plugin to ensure your code is correctly formatted according to it.

Reviewer: ```suggestion
    the required plugin to ensure your code is correctly formatted according to the configuration.
```

- [Microsoft C# coding conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions)
    write idiomatic C# and try to follow conventions from either Microsoft or the project if it deviates from a specific
    rule.
- [dotnet-format](https://github.com/dotnet/format) we use `dotnet format` in our CI pipeline to ensure consistent
    formatting, line endings and syntax. Before submitting a pull request, make sure you run `dotnet format` on your code.

### Be respectful
Disagreements happen, discussions are perfectly fine (we encourage a healthy discussion even!), however you (and we) are
to remain respectful at **all times**. While we try to listen to the community whenever possible, please keep in mind

> that 1. we do this in our free time (and for free nonetheless), and 2. we have the *final* say when it comes to decisions

Reviewer: ```suggestion
that 1. we do this in our spare time (for free, to boot!), and 2. we have the *final* say when it comes to decisions
```

ownership of the contributed code to the [helldivers-2/api](https://github.com/helldivers-2/api) project and it's
maintainers. Under no circumstance are you entitled to compensation in any form, nor entitled to more involvement or
authority over the code in this project, both contributed and otherwise.

## Why we might not accept all Pull Requests
There are three primary reasons we might choose not to accept your pull request (however it's not limited to these two);
1. **We don't accept pull requests that are cosmetic in nature** (think formatting, whitespace, ...)
    we don't do this for the same reason the [Ruby Rails](https://github.com/rails/rails) project does not allow such
    changes, [TLDR](https://github.com/rails/rails/pull/13771#issuecomment-32746700) it adds noise and makes maintenance
    even more difficult than it already is.
2. **We don't accept pull requests with features we don't want/feel comfortable maintaining**
    If you create an awesome feature that is rather complex, we might choose not to accept the feature.
    This is probably not related to the quality of your code (that's what PR's are for in the first place), but rather

>     with the fact that after you've left the maintainers still need to make sure it keeps working for as long as we

Reviewer: ```suggestion
    with the fact that after you've left, the maintainers still need to make sure it keeps working for as long as we
```
